src/interfaces/etcd.py

```diff
@@ -0,0 +1,74 @@
+"""ETCD Interface for AMS charm."""
+# -*- coding: utf-8 -*-
+#
+#  Copyright 2024 Canonical Ltd.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+
+import ops
+
+logger = logging.getLogger(__name__)
+
+
+class Available(ops.EventBase):
+    """Event emitted when a new client is registered."""
+
+
+class ETCDEvents(ops.ObjectEvents):
+    """Event wrapper for ETCD events."""
+
+    available = ops.EventSource(Available)
+
+
+class ETCDEndpointConsumer(ops.framework.Object):
```

Comment:
Is there an existing interface implementation we can reuse already?

Reply:
The etcd charm itself is in reactive and searched around a bit for a charm lib covering this, but did not find one yet.